src/graph/serializers/serialize_schema.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018-2019 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Apache License, Version 2.0,
+ * modified with the Commons Clause restriction.
+ */
+
+#include "serialize_schema.h"
+#include "../../util/rmalloc.h"
+
+/* Deserialize schema */
+Schema* RdbLoadUnifiedSchema(RedisModuleIO *rdb) {
+  /* Format:
+   * id
+   * name
+   * #attributes
+   * attributes
+   */
+
+  int id = RedisModule_LoadUnsigned(rdb);
+  char *name = RedisModule_LoadStringBuffer(rdb, NULL);
+  Schema *s = Schema_New(name, id);
+
+  size_t len = 0;
+  uint64_t attrCount = RedisModule_LoadUnsigned(rdb);
+
+  for(int i = 0; i < attrCount; i++) {
+    // Load attribute string from RDB file.
+    char *attr = RedisModule_LoadStringBuffer(rdb, &len);
+
+    // Add the string directly to the schema triemap using the RDB-given length.
+    Attribute_ID *pAttribute_id = NULL;
+    pAttribute_id = malloc(sizeof(Attribute_ID));
+    *pAttribute_id = i;
+
+    TrieMap_Add(s->attributes, attr, len, pAttribute_id, TrieMap_NOP_REPLACE);
+
+    // Immediately free the string, as the schema does not reference it.
+    RedisModule_Free(attr);
+  }
+
+  return s;
+}
+
+Schema* RdbLoadSchema(RedisModuleIO *rdb, SchemaType type) {
+  /* Format:
+   * id
+   * name
+   * #attributes
+   * attributes
+   */
+
+  int id = RedisModule_LoadUnsigned(rdb);
+  char *name = RedisModule_LoadStringBuffer(rdb, NULL);
+  Schema *s = Schema_New(name, id);
+
+  size_t len = 0;
+  uint64_t attrCount = RedisModule_LoadUnsigned(rdb);
+
+  char attribute[1024];
+
+  for(int i = 0; i < attrCount; i++) {
+    // Load attribute string from RDB file.
+    char *attr = RedisModule_LoadStringBuffer(rdb, &len);    
+    
+    // TODO: make sure we do not overflow.
```

Comment:
See my comment in `serialize_graph.c`, I feel like we can smarter with this behavior. Saving the strings with an extra byte, for example, would let us avoid the `memcpy` here (we can even allow it to be a garbage byte and NULL-assign it here).

Reply:
I've tried updating Schema_AddAttribute to
TrieMap_Add(s->attributes,
                        (char*)attribute,
                        strlen(attribute)+1,
                        pAttribute_id,
                        TrieMap_NOP_REPLACE);

Made some additional modifications, but couldn't get our test suite pass (10 minutes investment), I don't want to spend additional time on that, feel free to give this a try.